### PR TITLE
Use pulse configuration in fake Bogota, Rome, Manila and Santiago 

### LINKED ISSUE #7685: FakeBogota, FakeManila, FakeRome, and FakeSantiago can be a FakePulseBackend
### Environment

- **Qiskit Terra version**: 0.19.2
- **Python version**: 
- **Operating system**: 

### What is happening?

fake_bogota, fake_manila.py, fake_rome.py, and fake_santiago.py don't refer to its defs_bogota, defs_manila.json, defs_rome.json, and defs_santiago.json file and it could.
https://github.com/Qiskit/qiskit-terra/tree/main/qiskit/test/mock/backends/bogota
https://github.com/Qiskit/qiskit-terra/tree/main/qiskit/test/mock/backends/manila
https://github.com/Qiskit/qiskit-terra/tree/main/qiskit/test/mock/backends/rome
https://github.com/Qiskit/qiskit-terra/tree/main/qiskit/test/mock/backends/santiago

### How can we reproduce the issue?

https://github.com/Qiskit/qiskit-terra/tree/main/qiskit/test/mock/backends/bogota
https://github.com/Qiskit/qiskit-terra/tree/main/qiskit/test/mock/backends/manila
https://github.com/Qiskit/qiskit-terra/tree/main/qiskit/test/mock/backends/rome
https://github.com/Qiskit/qiskit-terra/tree/main/qiskit/test/mock/backends/santiago

### What should happen?

change line 21 and 30

### Any suggestions?

_No response_

### PR DESCRIPTION
**Pull request by omarcostahamido**
### Summary

this fixes #7685

### CODE REVIEW

File: qiskit/test/mock/backends/bogota/fake_bogota.py [hunk 1]
```diff
@@ -15,23 +15,24 @@
...
 
     dirname = os.path.dirname(__file__)
     conf_filename = "conf_bogota.json"
     props_filename = "props_bogota.json"
+    defs_filename = "defs_bogota.json"
     backend_name = "fake_bogota"
 
 
-class FakeLegacyBogota(fake_qasm_backend.FakeQasmLegacyBackend):
+class FakeLegacyBogota(fake_pulse_backend.FakePulseBackend):
```
Comment (jakelishman) [inline qiskit/test/mock/backends/bogota/fake_bogota.py:31]: This needs to be a type of legacy backend.

**Review by jakelishman — COMMENTED** [review]
Thanks for your interest in helping out on Qiskit!

If updating the one comment I've left is all that's required to fix the test failures, I'm happy to accept the PR.  If it needs more work and care, or if the definitions are too out-of-date, I'm not super keen to put in the reviewer time, though - change for change's sake isn't great for maintainability, and unless there's buggy behaviour, it's better for us to keep consistency in our public API.

If the tests pass after swapping `LegacyBogota` to be a `PulseLegacyBackend`, this will need a one-line "feature" release note saying that `FakeBogota`, etc now contain some sample pulse calibration data.

File: qiskit/test/mock/backends/bogota/fake_bogota.py (hunk 1 above)
Comment (omarcostahamido) [reply qiskit/test/mock/backends/bogota/fake_bogota.py:31]: Great catch. Thank you. I'm on to it.

**Comment by jakelishman** [comment]
@omarcostahamido: this has all gone through successfully - if you'd like to add a short release note, I'll get this merged.

**Comment by omarcostahamido** [comment]
@jakelishman Thank you. I've added the release notes and updated the branch.

(feels good to not have spent time studying this and just produce 🗑️ ...)

File: releasenotes/notes/bogota-manila-rome-santiago-as-fakepulsebackends-2907dec149997a27.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+prelude: >
+    FakeBogota, FakeManila, FakeRome, and FakeSantiago can now be used as fake backends in Pulse experiments.
```
Comment (mtreinish) [inline releasenotes/notes/bogota-manila-rome-santiago-as-fakepulsebackends-2907dec149997a27.yaml:3]: You don't need a prelude section here. This section is reserved for the high level overview/highlights of the release. The contents of all the prelude sections are merged together into a single text section so we typically just write a signle release note with this section at release time.

```suggestion
```

File: releasenotes/notes/bogota-manila-rome-santiago-as-fakepulsebackends-2907dec149997a27.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+features:
+  - |
+    FakeBogota, FakeManila, FakeRome, and FakeSantiago can now be used as fake backends in Pulse experiments.
```
Comment (mtreinish) [inline releasenotes/notes/bogota-manila-rome-santiago-as-fakepulsebackends-2907dec149997a27.yaml:6]:
```suggestion
    The ``FakeBogota``, ``FakeManila``, ``FakeRome``, and ``FakeSantiago`` fake backends which can be found
    in the ``qiskit.test.mock`` module can now be used as backends in Pulse experiments as they now include the
    a :class:`~qiskit.providers.models.PulseDefaults` created from a snapshot of the IBM Quantum's properties.
```

**Review by mtreinish — COMMENTED** [review]
LGTM, just some small suggestions in the release note then I think this is ready to merge

File: releasenotes/notes/bogota-manila-rome-santiago-as-fakepulsebackends-2907dec149997a27.yaml (hunk 2 above)
Comment (omarcostahamido) [reply releasenotes/notes/bogota-manila-rome-santiago-as-fakepulsebackends-2907dec149997a27.yaml:3]: Ah ok. Thank you for clarifying that @mtreinish !

File: releasenotes/notes/bogota-manila-rome-santiago-as-fakepulsebackends-2907dec149997a27.yaml (hunk 3 above)
Comment (omarcostahamido) [reply releasenotes/notes/bogota-manila-rome-santiago-as-fakepulsebackends-2907dec149997a27.yaml:4]: Ok thank you so much for the added articulation.
Question: where can I learn about this notation you used here
>:class:`~qiskit.providers.models.PulseDefaults`

Meaning, how should I know that I need to write :class: and the "~" ?
@mtreinish

**Comment by omarcostahamido** [comment]
Ok thank you so much for the added articulation @mtreinish 
Question: where can I learn about this notation you used here
>:class:`~qiskit.providers.models.PulseDefaults`

Meaning, how should I know that I need to write :class: and the "~" ?

**Comment by mtreinish** [comment]
It's the sphinx syntax for cross-references, which is documented here: https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing-python-objects (we don't need the `:py` prefix here as it's all python)

**Comment by omarcostahamido** [comment]
Thank you for the reference @mtreinish !

All checks have passed ✅ 

😄

File: releasenotes/notes/bogota-manila-rome-santiago-as-fakepulsebackends-2907dec149997a27.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+features:
+  - |
+    The ``FakeBogota``, ``FakeManila``, ``FakeRome``, and ``FakeSantiago`` fake backends which can be found
+    in the ``qiskit.test.mock`` module can now be used as backends in Pulse experiments as they now include the
```
Comment (jakelishman) [inline releasenotes/notes/bogota-manila-rome-santiago-as-fakepulsebackends-2907dec149997a27.yaml:5]:
```suggestion
    in the ``qiskit.test.mock`` module can now be used as backends in Pulse experiments as they now include
```

**Comment by HuangJunye** [comment]
@omarcostahamido I am working on #7643 and found this problem. Thanks for fixing this!

**Comment by omarcostahamido** [comment]
@HuangJunye my pleasure :)
Looking forward to see the work you're doing there when it gets merged as well.
It is funny that even fake backends have different generations now.
Perhaps, like the real QC hardware,  they could also have generation codenames: mockingbirdv1, mockingbirdv2, etc 😄 
I just created my first custom>merged notification on your PR (was about to ask how to do this), I might need to update my code then. 🙏

**Pull request merged**